### ISSUE TITLE
New message: FOP_ValidateFOP

### ISSUE THREAD

**Issue by bimusiek** [amabnl/amadeus-ws-client#76]
Hey, 
what do you think about implementing `FOP_ValidateFOP`? 😄 

It is mostly useful for non-iata agencies that issue tickets by sending it to consolidator. 
In our case, we get CC declined from time to time and process after CC is declined is a hell. 
Communicating with customer and with consolidator is a lot of hassle.

With `FOP_ValidateFOP`, before putting PNR on issueing queue, we can first "reserve" money on the CC proving it can be charged, so there is never a case when CC is declined as we know before trying to issue ticket.

**Comment by DerMika** [comment]
Do you have an idea which options you'll be needing? Can you give me some sample message that would include all the options you need? 

I'm asking mainly because there's only 2 small samples in the Extranet docs, and I don't mind doing that message to allow those options, but the request message XSD also includes the entire `fopGroup` structure - which would be a lot more work to implement :)

**Comment by bimusiek** [comment]
Right now we have to validate one FOP element so this
```xml
<message agency="1A" release="2" type="TFOPVQ" version="15">
    <fopGroup>
        <fopReference>
            <reference>
                <qualifier>FPT</qualifier>
                <number>12</number>
            </reference>
        </fopReference>
    </fopGroup>
</message>
```
should be enough. I don't see first examples though, so if you see easier way in docs to validate all FOPs this will also work. 

We are already talking with our consolidator to see if we can do Split Payments, as per checking Amadeus I see that `FOP_CreateFormOfPayment` endpoint allows it, so in the future, we will need to be able to validate many FOPs. 

Can be done with validating each of them separately, but we would also be happy by validating all of them together.

**Comment by DerMika** [comment]
Ok, I'll see what I can do. Perhaps this week, but no promises.

**Comment by bimusiek** [comment]
Hey, will you have time to implement basic message this week? 😉 
If not, I will start working on it. I have a lot to catch up with PHP, I worked on it last time when they just released 5.0 🙈

**Comment by DerMika** [comment]
Work being done in #86

@bimusiek did you get a chance to look at why your unittest is failing?

**Comment by bimusiek** [comment]
Unfortunately not, but I will check it out this weekend. 
I will run it in Docker with some linux distro and will see if it fails.

**Comment by DerMika** [comment]
Fixed by yourself in #86, so I'll close this one. Thanks again ;)
